src/lightning/pytorch/__init__.py

```diff
@@ -26,11 +26,20 @@
 from lightning.pytorch.callbacks import Callback  # noqa: E402
 from lightning.pytorch.core import LightningDataModule, LightningModule  # noqa: E402
 from lightning.pytorch.trainer import Trainer  # noqa: E402
+from lightning.pytorch.utilities.data import LightningDataset, S3LightningDataset  # noqa: E402
 
 # this import needs to go last as it will patch other modules
 import lightning.pytorch._graveyard  # noqa: E402, F401  # isort: skip
 
-__all__ = ["Trainer", "LightningDataModule", "LightningModule", "Callback", "seed_everything"]
+__all__ = [
+    "Trainer",
+    "LightningDataModule",
+    "LightningModule",
+    "S3LightningDataset",
```

Comment:
```suggestion
```
why need to have S3 at the top level, will we later include also others like GCP etc.?

Reply:
yes that's the idea